packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -44,16 +54,46 @@ const ColorPanel = ( {
 		colorSettings={ Object.values( colorSettings ) }
 		{ ...colorPanelProps }
 	>
-		{ contrastCheckerProps &&
-			map( components, ( ( Component, key ) => (
-				<ContrastChecker
-					key={ key }
-					textColor={ colorSettings[ key ].value }
-					{ ...contrastCheckerProps }
-				/>
-			) ) ) }
+		{ contrastCheckers &&
+			( Array.isArray( contrastCheckers ) ?
```

Comment:
The code between the array and non-array version is very similar could we do no arrayCast and simplify the logic?

Reply:
It would be very confusing.

We are mapping over 2 different things, `contrastCheckers` vs. `colorSettings`, and values are handled a bit differently.